LanguageFeatures/Patterns/constant_A03_t03.dart

```diff
@@ -23,117 +24,72 @@
 /// literals explicitly marked const. Likewise with set and map literals versus
 /// map patterns.
 ///
-/// @description Check list, map and set literals with type argument specified
-/// in constant patterns
+/// @description Check list, map and set literals in constant patterns. Test
+/// switch expression
 /// @author sgrekhov22@gmail.com
 
 // SharedOptions=--enable-experiment=patterns
 
 import "../../Utils/expect.dart";
 
 String testList(List value) {
-  switch (value) {
-    case const <int>[1, 2]:
-      return "<int>[1, 2]";
-    case const <dynamic>[1, 2]:
-      return "<dynamic>[1, 2]";
-    case const <String>["3", "4"]:
-      return "<String>['3', '4']";
-    case const <dynamic>["3", "4"]:
-      return "<dynamic>['3', '4']";
-    case const <double>[]:
-      return "<double>[]";
-    case const <dynamic>[]:
-      return "<dynamic>[]";
-    default:
-      return "default";
-  }
+  return switch (value) {
+    case const [1, 2] => "[1, 2]";
+    case const [1, -2] => "[1, -2]";
+    case const ["3", "4"] => "['3', '4']";
+    case const [true, true] => "[true, true]";
+    case const [] => "[]";
+    default => "default";
+  };
 }
 
 String testMap(Map value) {
-  switch (value) {
-    case const <int, int>{1: 2}:
-      return "<int, int>{1: 2}";
-    case const <dynamic, dynamic>{1: 2}:
-      return "<dynamic, dynamic>{1: 2}";
-    case const <String, num>{'answer': 42}:
-      return "<String, num>{'answer': 42}";
-    case const <dynamic, dynamic>{'answer': 42}:
-      return "<dynamic, dynamic>{'answer': 42}";
-    case const <String, Object>{'true': true}:
-      return "<String, Object>{'true': true}";
-    case const <dynamic, dynamic>{'true': true}:
-      return "<dynamic, dynamic>{'true': true}";
-    case const <String, int>{}:
-      return "<String, int>{}";
-    case const <dynamic, dynamic>{}:
-      return "<dynamic, dynamic>{}";
-    default:
-      return "default";
-  }
+  return switch (value) {
+    case const {1: 2} => "{1: 2}";
+    case const {'answer': -42} => "{'answer': -42}";
+    case const {'true': true} => "{'true': true}";
+    case const {} => "{}";
+    default => "default";
+  };
 }
 
 String testSet(Set value) {
-  switch (value) {
-    case const <num>{1, 2, 3}:
-      return "<num>{1, 2, 3}";
-    case const <dynamic>{1, 2, 3}:
-      return "<dynamic>{1, 2, 3}";
-    case const <String>{'1', '2', '3'}:
-      return "<String>{'1', '2', '3'}";
-    case const <dynamic>{'1', '2', '3'}:
-      return "<dynamic>{'1', '2', '3'}";
-    case const <double>{}:
-      return "<double>{}";
-    case const <dynamic>{}:
-      return "<dynamic>{}";
-    default:
-      return "default";
-  }
+  return switch (value) {
+    case const {1, 2, -3} => "{1, 2, -3}";
+    case const {'1', '2', '3'} => "{'1', '2', '3'}";
+    case const {} => "{}";
+    default => "default";
+  };
```

Comment:
These guys would then have to use `()` (or the use of `const (...)` is already tested elsewhere and doesn't belong here, we just need to check that this other location testing `const (...)` does include cases with sets).